wsagent/che-core-api-languageserver/src/main/java/org/eclipse/che/api/languageserver/registry/LanguageServerRegistryImpl.java

```diff
@@ -26,99 +25,117 @@
 import org.eclipse.lsp4j.ServerCapabilities;
 import org.eclipse.lsp4j.services.LanguageServer;
 
+import java.io.File;
 import java.net.URI;
 import java.util.ArrayList;
-import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Map;
 import java.util.Set;
 import java.util.concurrent.ConcurrentHashMap;
 import java.util.stream.Collectors;
 
-import static com.google.common.io.Files.getFileExtension;
-import static org.eclipse.che.api.languageserver.shared.ProjectExtensionKey.createProjectKey;
+import static org.eclipse.che.api.languageserver.shared.ProjectLangugageKey.createProjectKey;
 
 @Singleton
 public class LanguageServerRegistryImpl implements LanguageServerRegistry, ServerInitializerObserver {
-    public final static String PROJECT_FOLDER_PATH = "/projects";
-
-    /**
-     * Available {@link LanguageServerLauncher} by extension.
-     */
-    private final ConcurrentHashMap<String, List<LanguageServerLauncher>> extensionToLauncher;
+    public final static String                        PROJECT_FOLDER_PATH = "/projects";
+    private final List<LanguageDescription>           languages           = new ArrayList<>();
+    private final Map<String, LanguageServerLauncher> launchers           = new HashMap<>();
 
     /**
      * Started {@link LanguageServer} by project.
      */
-    private final ConcurrentHashMap<ProjectExtensionKey, LanguageServer> projectToServer;
+    private final ConcurrentHashMap<ProjectLangugageKey, LanguageServer> projectToServer;
 
     private final Provider<ProjectManager> projectManagerProvider;
     private final ServerInitializer        initializer;
 
     @Inject
-    public LanguageServerRegistryImpl(Set<LanguageServerLauncher> languageServerLaunchers,
-                                      Provider<ProjectManager> projectManagerProvider,
-                                      ServerInitializer initializer) {
+    public LanguageServerRegistryImpl(Set<LanguageServerLauncher> languageServerLaunchers, Set<LanguageDescription> languages,
+                                      Provider<ProjectManager> projectManagerProvider, ServerInitializer initializer) {
+        this.languages.addAll(languages);
+        this.launchers.putAll(languageServerLaunchers.stream()
+                        .collect(Collectors.toMap(LanguageServerLauncher::getLanguageId, launcher -> launcher)));
```

Comment:
if languages and launchers are initialized there, do we need to initialize the fields and use addAll/putAll after instead of just using the new directly there

```
this.languages = new ArrayList<>(languages);
```
vs
```
List list = new ArrayList<>()   (when declaring field)
list.addAll(myElements)  (in constructor)
```